docs/source/accelerators/hpu.rst

```diff
@@ -0,0 +1,204 @@
+
+.. _hpu_accelerator:
+
+HPU accelerator
+---------------
+
+The :code:`devices=1` with :code:`accelerator="hpu"` parameters in the trainer class enables the Habana backend.
+
+
+.. _single_device_strategy:
+
+Training on Single HPU
+----------------------
+
+The :code:`devices=1` and :code:`accelerator="hpu"` with :code:`strategy=SingleHPUStrategy(device=torch.device("hpu"))` parameter in the trainer class enables the Habana backend for single Gaudi training.
+
+
+.. _parallel_device_strategy:
+
+Distributed Training
+---------------------
+
+
+The :code:`devices=8` and :code:`accelerator="hpu"` with :code:`strategy=HPUParallelStrategy( parallel_devices=[torch.device("hpu")] * devices)`  parameter in the trainer class enables the Habana backend for distributed training with 8 Gaudis.
+
+The Habana parallel device strategy is based on DDP strategy with the addition of  Habana's collective communication library (HCCL) to support scale-up within a node and scale-out across multiple nodes.
+
+
+.. _mixed_precision_plugin:
+
+Mixed Precision Plugin
+----------------------
+
+The :code:`precision=16` and a :code:`hmp_params` parameter in the trainer class enables the Habana plugin for mixed precision using the Habana Mixed Precision (HMP) package.
+
+You can execute the ops in FP32 or BF16 precision. The HMP package modifies the python operators to add the appropriate cast operations for the arguments before execution.
+The default settings enable users to easily enable mixed precision training with minimal code.
+
+In addition to the default settings in HMP,  users also have the option of overriding these defaults and providing their own BF16 and FP32 operator lists.
+
+For more details, please refer `<https://docs.habana.ai/en/latest/PyTorch_User_Guide/PyTorch_User_Guide.html#pytorch-mixed-precision-training-on-gaudi>`_.
+
+
+.. _pytorch_lightning_examples:
+
+Getting Started with Lightning on Gaudi
+=======================================
+
+This section describes how to train models using PyTorch Lightning with Habana Gaudi.
+
+More Lightning HPU examples can be found in  pl_examples (`<https://github.com/PyTorchLightning/pytorch-lightning/pl_examples/hpu_examples/ >`)
+
+Enabling Lightning with Single Gaudi HPU
+----------------------------------------
+
+The below snippet shows an example model using MNIST with single Habana Gaudi device:
+
+.. code-block:: python
+
+    import habana_frameworks.torch.core as htcore
+
+
+    class LitClassifier(pl.LightningModule):
+        def __init__(self):
+            super(LitClassifier, self).__init__()
+
+        ...
+
+
+    # Init our model
+    model = LitClassifier()
+
+    # Init DataLoader from MNIST Dataset
+    dm = MNISTDataModule(batch_size=batch_size)
+
+    ...
+
+    num_hpus = 1
+
+    # enable HPU strategy for single device, with mixed precision using default HMP settings
+    hpustrat_1 = SingleHPUStrategy(device=torch.device("hpu"), precision_plugin=HPUPrecisionPlugin(precision=16))
+
```

Comment:
is this needed? shouldn't this be equivalent?
```py
trainer = pl.Trainer(accelerator="hpu", devices=1, precision=16) 
```

Reply:
To expose hmp_params, there was a request to create backend specific plugin. Hence the change